ingenialink/canopen/network.py

```diff
@@ -356,375 +359,323 @@ def load_firmware(
+
+        Args:
+            initial_status: program control initial status
+            servo: target drive
+            callback_status_msg: Subscribed callback function for the status message
+
+        Raises:
+            ILFirmwareLoadError: Program control status does not change
+
+        """
+        if callback_status_msg:
+            callback_status_msg("Setting up drive")
+        target_status_list = [
+            PROG_CTRL_STATE_START,
+            PROG_CTRL_STATE_STOP,
+            PROG_CTRL_STATE_CLEAR,
+            PROG_CTRL_STATE_FLASH,
+        ]
+        status_index = target_status_list.index(initial_status) + 1
+        logger.info("Clearing program...")
+        for target_status_index in range(status_index, len(target_status_list)):
+            target_status = target_status_list[target_status_index]
+            servo.write(PROG_STAT_1, target_status, subnode=0)
+            if not self.__wait_for_register_value(servo, 0, PROG_STAT_1, target_status):
+                raise ILFirmwareLoadError(f"Error setting program control to 0x{target_status:X}")
+
+    @staticmethod
+    def __wait_for_register_value(
+        servo: CanopenServo,
+        subnode: int,
+        register: CanopenRegister,
+        expected_value: Union[int, float, str],
+    ) -> bool:
+        """Waits for the register to reach a value.
+
+        Args:
+            servo: Instance of the servo to be used.
+            subnode: Target subnode.
+            register: Register to be read.
+            expected_value: Expected value for the given register.
+
+        Returns:
+            True if values is reached, else False
+        """
+        logger.debug(f"Waiting for register {register} to return <{expected_value}>")
+        num_tries = 0
+        value = None
+        while num_tries < POLLING_MAX_TRIES:
+            with contextlib.suppress(ILError):
+                value = servo.read(register, subnode=subnode)
+            if value == expected_value:
+                logger.debug(f"Success. Read value {value}. Num tries {num_tries}")
+                return True
+            num_tries += 1
+            logger.debug(f"Trying again {num_tries}. value {value}.")
+            sleep(1)
+        return False
+
+    @staticmethod
+    def __program_control_to_stop(
+        servo: CanopenServo, callback_status_msg: Optional[Callable[[str], None]]
+    ) -> None:
+        """Change program control status to stop.
+
+        Args:
+            servo: target drive
+            callback_status_msg: Subscribed callback function for the status message
+
+        Raises:
+            ILFirmwareLoadError: Drive does not respond
+
+        """
+        if callback_status_msg:
+            callback_status_msg("Flashing firmware")
+        logger.info("Flashing firmware")
+        with contextlib.suppress(ILError):
+            servo.write(PROG_STAT_1, PROG_CTRL_STATE_STOP, subnode=0)
+        servo.node.nmt.start_node_guarding(5)
```

Comment:
You could use the key of the argument to make clarifier what this 5 is. What do you think?